### PR TITLE
fix few resource leaks

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

**Comment by hardening** [comment]
@freerdp-bot can you test please ?

File: libfreerdp/utils/http.c [hunk 1]
```diff
@@ -126,7 +126,10 @@ BOOL freerdp_http_request(const char* url, const char* body, long* status_code,
 	{
 		blen = strlen(body);
 		if (winpr_asprintf(&headers, &size, post_header_fmt, path, hostname, blen) < 0)
+		{
+			free(hostname);
```
Comment (akallabeth) [inline libfreerdp/utils/http.c:130]: same `free` missing in else branch

**Review by akallabeth — CHANGES_REQUESTED** [review]
LGTM, just one more free required

File: libfreerdp/utils/http.c (hunk 1 above)
Comment (chipitsine) [reply libfreerdp/utils/http.c:130]: thanks, added

**Review by akallabeth — APPROVED** [review]
LGTM, thank you!

**Pull request merged**